File: app/lib/customer-auth.ts
import { toast } from 'sonner';
import { API_BASE_URL } from './config';

export interface CustomerUser {
  id: number;
  email: string;
  first_name: string;
  last_name: string;
  email_verified: boolean;
  is_active: boolean;
  created_at: string;
  last_login?: string;
  updated_at?: string;
}

export interface CustomerLoginCredentials {
  email: string;
  password: string;
  remember_me?: boolean;
}

export interface CustomerRegisterCredentials {
  email: string;
  password: string;
  first_name: string;
  last_name: string;
}

export interface CustomerAuthTokens {
  access_token: string;
  refresh_token: string;
  access_expires_in: number;
  refresh_expires_in: number;
}

export interface CustomerAuthResponse {
  message: string;
  user: CustomerUser;
  tokens: CustomerAuthTokens;
}

export interface CustomerProfileResponse {
  user: CustomerUser;
}

export interface CustomerRefreshResponse {
  message: string;
  tokens: CustomerAuthTokens;
}

export interface CsrfResponse {
  csrf_token: string;
}

class CustomerAuthAPI {
  private baseURL: string;
  private csrfToken: string | null = null;
  private retryAttempts = 3;
  private retryDelay = 1000; // 1 second

  constructor() {
    this.baseURL = API_BASE_URL;
    console.log('CustomerAuthAPI initialized with URL:', this.baseURL);
  }

  private async request<T>(
    endpoint: string,
    options: RequestInit = {},
    retryCount = 0
  ): Promise<T> {
    const url = `${this.baseURL}${endpoint}`;
    console.log('Making customer auth request to:', url);
    
    const config: RequestInit = {
      headers: {
        'Content-Type': 'application/json',
        ...options.headers,
      },
      credentials: 'include', // Include cookies for CSRF
      mode: 'cors', // Explicitly set CORS mode
      ...options,
    };

    // Add CSRF token if available
    if (this.csrfToken && options.method !== 'GET') {
      config.headers = {
        ...config.headers,
        'X-CSRF-Token': this.csrfToken,
      };
    }

    try {
      const response = await fetch(url, config);
      
      console.log('Response received:', {
        status: response.status,
        statusText: response.statusText,
        headers: Object.fromEntries(response.headers.entries())
      });
      
      if (!response.ok) {
        let errorMessage = `HTTP error! status: ${response.status}`;
        
        try {
          const errorData = await response.json();
          errorMessage = errorData.message || errorData.error || errorMessage;
        } catch (parseError) {
          // If we can't parse the error response, use the status text
          errorMessage = response.statusText || errorMessage;
        }
        
        // Handle specific error cases
        if (response.status === 429) {
          // Rate limited
          throw new Error('Too many attempts. Please try again later.');
        } else if (response.status === 403) {
          // CSRF token invalid
          await this.refreshCsrfToken();
          throw new Error('Session expired. Please try again.');
        } else if (response.status === 401) {
          // Unauthorized - token is invalid
          this.clearTokens();
          throw new Error('Authentication required. Please log in again.');
        } else if (response.status === 500) {
          // Server error - might be due to invalid user
          if (errorMessage.includes('User not found') || errorMessage.includes('Failed to update profile')) {
            this.clearTokens();
            throw new Error('Session expired. Please log in again.');
          }
        }
        
        throw new Error(errorMessage);
      }
      
      const data = await response.json();
      console.log('Response data:', data);
      
      // Check for specific error types in the response
      if (data.error === 'Account deleted') {
        throw new Error('This account has been deleted and cannot be accessed. Please create a new account.');
      }
      
      // For invalid credentials, provide a more helpful message
      if (data.error === 'Invalid credentials') {
        throw new Error('Invalid email or password. If you recently deleted your account, you will need to create a new one.');
      }
      
      return data;
      
    } catch (error: any) {
      console.error('Request failed:', error);
      
      // Enhanced error handling for network issues
      if (error.name === 'TypeError' && error.message.includes('fetch')) {
        if (retryCount < this.retryAttempts) {
          console.log(`Network error, retrying (${retryCount + 1}/${this.retryAttempts})...`);
          await this.delay(this.retryDelay * (retryCount + 1));
          return this.request(endpoint, options, retryCount + 1);
        } else {
          throw new Error('Network error: Unable to connect to the server. Please check your internet connection and try again.');
        }
      }
      
      // Handle CORS errors
      if (error.name === 'TypeError' && error.message.includes('CORS')) {
        throw new Error('CORS error: Unable to make request due to browser security restrictions.');
      }
      
      throw error;
    }
  }

  private isRetryableError(error: any): boolean {
    return error.name === 'TypeError' && error.message.includes('fetch');
  }

  private delay(ms: number): Promise<void> {
    return new Promise(resolve => setTimeout(resolve, ms));
  }

  async getCsrfToken(): Promise<string> {
    try {
      console.log('Getting CSRF token...');
      const response: CsrfResponse = await this.request('/customer/auth/csrf-token');
      this.csrfToken = response.csrf_token;
      console.log('CSRF token obtained:', this.csrfToken ? 'success' : 'failed');
      return response.csrf_token;
    } catch (error) {
      console.error('Failed to get CSRF token:', error);
      throw error;
    }
  }

  private async refreshCsrfToken(): Promise<void> {
    try {
      await this.getCsrfToken();
    } catch (error) {
      console.error('Failed to refresh CSRF token:', error);
    }
  }

  async register(credentials: CustomerRegisterCredentials): Promise<CustomerAuthResponse> {
    try {
      console.log('Starting registration process...');
      
      // Get CSRF token first
      await this.getCsrfToken();
      
      console.log('Making registration request...');
      const response = await this.request<CustomerAuthResponse>('/customer/auth/register', {
        method: 'POST',
        body: JSON.stringify(credentials),
      });
      
      console.log('Registration successful');
      return response;
    } catch (error) {
      console.error('Registration error:', error);
      throw error;
    }
  }

  async login(credentials: CustomerLoginCredentials): Promise<CustomerAuthResponse> {
    try {
      console.log('Starting login process...');
      
      const response = await this.request<CustomerAuthResponse>('/customer/auth/login', {
        method: 'POST',
        body: JSON.stringify(credentials),
      });
      
      console.log('Login successful');
      return response;
    } catch (error) {
      console.error('Login error:', error);
      throw error;
    }
  }

  async logout(): Promise<{ message: string }> {
    try {
      const accessToken = this.getAccessToken();
      if (!accessToken) {
        return { message: 'Already logged out' };
      }

      return await this.request<{ message: string }>('/customer/auth/logout', {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${accessToken}`,
        },
      });
    } catch (error) {
      console.error('Logout error:', error);
      throw error;
    }
  }

  async getProfile(): Promise<CustomerProfileResponse> {
    try {
      const accessToken = this.getAccessToken();
      if (!accessToken) {
        throw new Error('No access token available');
      }

      return await this.request<CustomerProfileResponse>('/customer/auth/profile', {
        method: 'GET',
        headers: {
          'Authorization': `Bearer ${accessToken}`,
        },
      });
    } catch (error) {
      console.error('Get profile error:', error);
      // If we get a 401 or 500 error, the token is likely invalid
      if (error.message.includes('Authentication required') || 
          error.message.includes('Session expired') ||
          error.message.includes('User not found')) {
        this.clearTokens();
      }
      throw error;
    }
  }

  async updateProfile(data: Partial<CustomerUser>): Promise<{ message: string; user: CustomerUser }> {
    try {
      const accessToken = this.getAccessToken();
      
      if (!accessToken) {
        throw new Error('No access token available');
      }

      return await this.request<{ message: string; user: CustomerUser }>('/customer/auth/profile', {
        method: 'PUT',
        headers: {
          'Authorization': `Bearer ${accessToken}`,
        },
        body: JSON.stringify(data),
      });
    } catch (error) {
      console.error('Update profile error:', error);
      // If we get a 401 or 500 error, the token is likely invalid
      if (error.message.includes('Authentication required') || 
          error.message.includes('Session expired') ||
          error.message.includes('User not found')) {
        this.clearTokens();
      }
      throw error;
    }
  }

  async changePassword(currentPassword: string, newPassword: string): Promise<{ message: string }> {
    try {
      const accessToken = this.getAccessToken();
      if (!accessToken) {
        throw new Error('No access token available');
      }

      return await this.request<{ message: string }>('/customer/auth/change-password', {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${accessToken}`,
        },
        body: JSON.stringify({
          current_password: currentPassword,
          new_password: newPassword
        }),
      });
    } catch (error) {
      console.error('Change password error:', error);
      throw error;
    }
  }

  async refreshToken(refreshToken: string): Promise<CustomerRefreshResponse> {
    try {
      return await this.request<CustomerRefreshResponse>('/customer/auth/refresh', {
        method: 'POST',
        body: JSON.stringify({ refresh_token: refreshToken }),
      });
    } catch (error) {
      console.error('Token refresh error:', error);
      throw error;
    }
  }

  async deleteAccount(data: { reason?: string } = {}): Promise<{ message: string; note: string }> {
    try {
      const accessToken = this.getAccessToken();
      if (!accessToken) {
        throw new Error('No access token available');
      }

      return await this.request<{ message: string; note: string }>('/customer/auth/delete-account', {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${accessToken}`,
        },
        body: JSON.stringify(data),
      });
    } catch (error: any) {
      console.error('Delete account error:', error);
      
      // Check if the error is due to user already being deleted
      if (error.message.includes('User not found') || error.message.includes('already deleted')) {
        throw new Error('Your account has already been deleted. You will be logged out.');
      }
      
      // Check if the error is due to account already being deleted
      if (error.message.includes('Account already deleted')) {
        throw new Error('Your account has already been deleted. You will be logged out.');
      }
      
      throw error;
    }
  }

  async testDelete(): Promise<{ message: string; user_id: number; user_found: boolean; user_email: string | null }> {
    try {
      const accessToken = this.getAccessToken();
      if (!accessToken) {
        throw new Error('No access token available');
      }

      return await this.request<{ message: string; user_id: number; user_found: boolean; user_email: string | null }>('/customer/auth/test-delete', {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${accessToken}`,
        },
      });
    } catch (error) {
      console.error('Test delete error:', error);
      throw error;
    }
  }

  async exportData(): Promise<{ message: string; data: any }> {
    try {
      const accessToken = this.getAccessToken();
      if (!accessToken) {
        throw new Error('No access token available');
      }

      return await this.request<{ message: string; data: any }>('/customer/auth/export-data', {
        method: 'GET',
        headers: {
          'Authorization': `Bearer ${accessToken}`,
        },
      });
    } catch (error) {
      console.error('Export data error:', error);
      throw error;
    }
  }

  // Token management methods
  setTokens(tokens: CustomerAuthTokens): void {
    if (typeof window !== 'undefined') {
      localStorage.setItem('customer_access_token', tokens.access_token);
      localStorage.setItem('customer_refresh_token', tokens.refresh_token);
      localStorage.setItem('customer_token_expires', (Date.now() + tokens.access_expires_in * 1000).toString());
      localStorage.setItem('customer_refresh_expires', (Date.now() + tokens.refresh_expires_in * 1000).toString());
    }
  }

  getAccessToken(): string | null {
    if (typeof window !== 'undefined') {
      return localStorage.getItem('customer_access_token');
    }
    return null;
  }

  getRefreshToken(): string | null {
    if (typeof window !== 'undefined') {
      return localStorage.getItem('customer_refresh_token');
    }
    return null;
  }

  clearTokens(): void {
    if (typeof window !== 'undefined') {
      localStorage.removeItem('customer_access_token');
      localStorage.removeItem('customer_refresh_token');
      localStorage.removeItem('customer_token_expires');
      localStorage.removeItem('customer_refresh_expires');
    }
  }

  isTokenExpired(): boolean {
    if (typeof window !== 'undefined') {
      const expiresAt = localStorage.getItem('customer_token_expires');
      if (!expiresAt) return true;
      
      return Date.now() > parseInt(expiresAt);
    }
    return true;
  }

  isRefreshTokenExpired(): boolean {
    if (typeof window !== 'undefined') {
      const expiresAt = localStorage.getItem('customer_refresh_expires');
      if (!expiresAt) return true;
      
      return Date.now() > parseInt(expiresAt);
    }
    return true;
  }
}

// Export singleton instance
export const customerAuthApi = new CustomerAuthAPI();

File: backend/routes/customer_auth.py
from flask import Blueprint, request, jsonify, current_app
from models import db, CustomerUser
from utils.auth import (
    generate_tokens, verify_token, require_customer_auth, update_last_login, 
    validate_password, validate_email, check_rate_limit,
    generate_csrf_token, require_csrf
)
from datetime import datetime
import re

customer_auth_bp = Blueprint('customer_auth', __name__, url_prefix='/api/customer/auth')

@customer_auth_bp.route('/csrf-token', methods=['GET'])
def get_csrf_token():
    """Get CSRF token for form submission"""
    return jsonify({
        'csrf_token': generate_csrf_token()
    }), 200

@customer_auth_bp.route('/register', methods=['POST'])
def register():
    """Customer registration endpoint"""
    data = request.get_json()
    
    if not data:
        return jsonify({'error': 'No data provided'}), 400
    
    email = data.get('email', '').strip().lower()
    password = data.get('password', '')
    first_name = data.get('first_name', '').strip()
    last_name = data.get('last_name', '').strip()
    
    # Field validation
    if not email:
        return jsonify({'error': 'Email is required'}), 400
    
    if not password:
        return jsonify({'error': 'Password is required'}), 400
    
    if not first_name:
        return jsonify({'error': 'First name is required'}), 400
    
    if not last_name:
        return jsonify({'error': 'Last name is required'}), 400
    
    # Validate email format
    is_valid_email, error_message = validate_email(email)
    if not is_valid_email:
        return jsonify({'error': error_message}), 400
    
    # Validate password strength
    is_valid_password, error_message = validate_password(password)
    if not is_valid_password:
        return jsonify({'error': error_message}), 400
    
    try:
        # Check if user already exists
        existing_user = CustomerUser.query.filter_by(email=email).first()
        if existing_user:
            return jsonify({'error': 'Email already registered'}), 409
        
        # Create new user
        new_user = CustomerUser(
            email=email,
            first_name=first_name,
            last_name=last_name
        )
        new_user.set_password(password)
        
        # Save to database
        db.session.add(new_user)
        db.session.commit()
        
        # Generate tokens
        tokens = generate_tokens(new_user.id, new_user.email, 'customer', False)
        
        return jsonify({
            'message': 'Registration successful',
            'user': new_user.to_dict_public(),
            'tokens': tokens
        }), 201
        
    except Exception as e:
        current_app.logger.error(f"Registration error: {e}")
        db.session.rollback()
        return jsonify({
            'error': 'Registration failed',
            'message': 'Please try again later.'
        }), 500

@customer_auth_bp.route('/login', methods=['POST'])
def login():
    """Customer login endpoint"""
    data = request.get_json()
    
    if not data:
        return jsonify({'error': 'No data provided'}), 400
    
    email = data.get('email', '').strip().lower()
    password = data.get('password', '')
    remember_me = data.get('remember_me', False)
    
    # Field validation
    if not email:
        return jsonify({'error': 'Email is required'}), 400
    
    if not password:
        return jsonify({'error': 'Password is required'}), 400
    
    try:
        # Find user by email
        user = CustomerUser.query.filter_by(email=email).first()
        
        if not user:
            # Check if this email might belong to a deleted account
            # We can't directly look it up since we anonymize emails, but we can provide a helpful message
            return jsonify({'error': 'Invalid credentials'}), 401
        
        # Check if account is deleted
        if user.deleted_at is not None:
            return jsonify({
                'error': 'Account deleted',
                'message': 'This account has been deleted and cannot be accessed.'
            }), 401
        
        if not user.is_active:
            return jsonify({'error': 'Account is deactivated'}), 401
        
        # Check if account is locked
        if user.is_locked():
            remaining = user.get_lockout_remaining()
            return jsonify({
                'error': f'Account locked due to too many failed attempts. Please try again in {remaining} seconds.'
            }), 429
        
        # Check rate limiting
        is_allowed, error_message = check_rate_limit(user)
        if not is_allowed:
            return jsonify({'error': error_message}), 429
        
        # Verify password
        if not user.check_password(password):
            # Increment failed attempts
            user.increment_failed_attempts()
            
            # Check if account is now locked
            if user.is_locked():
                remaining = user.get_lockout_remaining()
                return jsonify({
                    'error': f'Account locked due to too many failed attempts. Please try again in {remaining} seconds.'
                }), 429
            
            return jsonify({'error': 'Invalid credentials'}), 401
        
        # Successful login - reset failed attempts
        user.reset_failed_attempts()
        
        # Generate tokens
        tokens = generate_tokens(user.id, user.email, 'customer', remember_me)
        
        # Update last login
        update_last_login(user.id)
        
        return jsonify({
            'message': 'Login successful',
            'user': user.to_dict_public(),
            'tokens': tokens
        }), 200
        
    except Exception as e:
        current_app.logger.error(f"Login error: {e}")
        return jsonify({
            'error': 'Login failed',
            'message': 'Please try again later.'
        }), 500

@customer_auth_bp.route('/logout', methods=['POST'])
@require_customer_auth
def logout():
    """Customer logout endpoint"""
    try:
        # In a real implementation, you might want to blacklist the token
        # For now, we'll just return success
        return jsonify({
            'message': 'Logout successful'
        }), 200
    except Exception as e:
        current_app.logger.error(f"Logout error: {e}")
        return jsonify({
            'error': 'Logout failed'
        }), 500

@customer_auth_bp.route('/profile', methods=['GET'])
@require_customer_auth
def get_profile():
    """Get customer profile"""
    try:
        user_id = request.user_id
        user = CustomerUser.query.get(user_id)
        
        if not user:
            return jsonify({'error': 'User not found'}), 404
        
        return jsonify({
            'user': user.to_dict_public()
        }), 200
    except Exception as e:
        current_app.logger.error(f"Get profile error: {e}")
        return jsonify({
            'error': 'Failed to get profile'
        }), 500

@customer_auth_bp.route('/profile', methods=['PUT'])
@require_customer_auth
def update_profile():
    """Update customer profile"""
    try:
        user_id = request.user_id
        user = CustomerUser.query.get(user_id)
        
        if not user:
            return jsonify({'error': 'User not found'}), 404
        
        data = request.get_json()
        
        if not data:
            return jsonify({'error': 'No data provided'}), 400
        
        # Update allowed fields
        if 'first_name' in data and data['first_name'].strip():
            user.first_name = data['first_name'].strip()
        
        if 'last_name' in data and data['last_name'].strip():
            user.last_name = data['last_name'].strip()
        
        db.session.commit()
        
        return jsonify({
            'message': 'Profile updated successfully',
            'user': user.to_dict_public()
        }), 200
        
    except Exception as e:
        current_app.logger.error(f"Update profile error: {e}")
        db.session.rollback()
        return jsonify({
            'error': 'Failed to update profile'
        }), 500

@customer_auth_bp.route('/change-password', methods=['POST'])
@require_customer_auth
def change_password():
    """Change customer password"""
    try:
        user_id = request.user_id
        user = CustomerUser.query.get(user_id)
        
        if not user:
            return jsonify({'error': 'User not found'}), 404
        
        data = request.get_json()
        
        if not data:
            return jsonify({'error': 'No data provided'}), 400
        
        current_password = data.get('current_password', '')
        new_password = data.get('new_password', '')
        
        if not current_password:
            return jsonify({'error': 'Current password is required'}), 400
        
        if not new_password:
            return jsonify({'error': 'New password is required'}), 400
        
        # Verify current password
        if not user.check_password(current_password):
            return jsonify({'error': 'Current password is incorrect'}), 401
        
        # Validate new password
        is_valid_password, error_message = validate_password(new_password)
        if not is_valid_password:
            return jsonify({'error': error_message}), 400
        
        # Update password
        user.set_password(new_password)
        db.session.commit()
        
        return jsonify({
            'message': 'Password changed successfully'
        }), 200
        
    except Exception as e:
        current_app.logger.error(f"Change password error: {e}")
        db.session.rollback()
        return jsonify({
            'error': 'Failed to change password'
        }), 500

@customer_auth_bp.route('/refresh', methods=['POST'])
def refresh_token():
    """Refresh access token using refresh token"""
    try:
        data = request.get_json()
        
        if not data or 'refresh_token' not in data:
            return jsonify({'error': 'Refresh token is required'}), 400
        
        refresh_token = data['refresh_token']
        
        # Verify refresh token
        payload, error = verify_token(refresh_token, 'refresh')
        if error or not payload:
            return jsonify({'error': 'Invalid refresh token'}), 401
        
        # Check if user exists
        user = CustomerUser.query.get(payload['user_id'])
        if not user or not user.is_active:
            return jsonify({'error': 'User not found or inactive'}), 401
        
        # Generate new tokens
        tokens = generate_tokens(user.id, user.email, 'customer', False)
        
        return jsonify({
            'message': 'Token refreshed successfully',
            'tokens': tokens
        }), 200
        
    except Exception as e:
        current_app.logger.error(f"Token refresh error: {e}")
        return jsonify({
            'error': 'Token refresh failed'
        }), 500

@customer_auth_bp.route('/delete-account', methods=['POST'])
@require_customer_auth
def delete_account():
    """Delete customer account - simplified version without password requirement"""
    try:
        current_app.logger.info("Delete account endpoint called")
        
        user_id = request.user_id
        current_app.logger.info(f"User ID from request: {user_id}")
        
        user = CustomerUser.query.get(user_id)
        current_app.logger.info(f"User found: {user is not None}")
        
        if not user:
            current_app.logger.warning(f"User not found for ID: {user_id}")
            return jsonify({'error': 'User not found'}), 404
        
        # Check if user is already deleted
        if user.deleted_at is not None:
            current_app.logger.info(f"User {user_id} is already deleted at {user.deleted_at}")
            return jsonify({
                'error': 'Account already deleted',
                'message': 'Your account has already been deleted and will be permanently removed after 30 days.'
            }), 400
        
        # Log the deletion request for audit purposes
        current_app.logger.info(f"Account deletion requested for user {user_id} ({user.email})")
        
        # Soft delete - mark as deleted but retain for legal compliance
        user.is_active = False
        user.deleted_at = datetime.utcnow()
        user.deletion_reason = 'User requested deletion'
        
        # Anonymize sensitive data (GDPR compliance)
        user.first_name = f"Deleted_{user_id}"
        user.last_name = "User"
        user.email = f"deleted_{user_id}@deleted.user"
        
        # Clear any sensitive data
        user.last_login = None
        
        current_app.logger.info("About to commit changes to database")
        
        # Commit the changes
        db.session.commit()
        
        # Log successful deletion
        current_app.logger.info(f"Account successfully deleted for user {user_id}")
        
        return jsonify({
            'message': 'Account deleted successfully',
            'note': 'Your data has been anonymized and will be permanently deleted after 30 days in accordance with our data retention policy.'
        }), 200
        
    except Exception as e:
        current_app.logger.error(f"Account deletion error: {e}")
        current_app.logger.error(f"Error type: {type(e)}")
        current_app.logger.error(f"Error details: {str(e)}")
        import traceback
        current_app.logger.error(f"Traceback: {traceback.format_exc()}")
        db.session.rollback()
        return jsonify({
            'error': 'Failed to delete account',
            'message': 'Please try again later or contact support.'
        }), 500

@customer_auth_bp.route('/test-delete', methods=['POST'])
@require_customer_auth
def test_delete():
    """Test endpoint to debug delete account functionality"""
    try:
        current_app.logger.info("Test delete endpoint called")
        
        user_id = request.user_id
        current_app.logger.info(f"User ID: {user_id}")
        
        user = CustomerUser.query.get(user_id)
        current_app.logger.info(f"User found: {user is not None}")
        
        if user:
            current_app.logger.info(f"User email: {user.email}")
            current_app.logger.info(f"User is_active: {user.is_active}")
            current_app.logger.info(f"User deleted_at: {user.deleted_at}")
        
        return jsonify({
            'message': 'Test successful',
            'user_id': user_id,
            'user_found': user is not None,
            'user_email': user.email if user else None
        }), 200
        
    except Exception as e:
        current_app.logger.error(f"Test delete error: {e}")
        import traceback
        current_app.logger.error(f"Traceback: {traceback.format_exc()}")
        return jsonify({
            'error': 'Test failed',
            'message': str(e)
        }), 500

@customer_auth_bp.route('/export-data', methods=['GET'])
@require_customer_auth
def export_user_data():
    """Export user data (GDPR compliance)"""
    try:
        user_id = request.user_id
        user = CustomerUser.query.get(user_id)
        
        if not user:
            return jsonify({'error': 'User not found'}), 404
        
        # Prepare user data for export
        user_data = {
            'user_id': user.id,
            'email': user.email,
            'first_name': user.first_name,
            'last_name': user.last_name,
            'created_at': user.created_at.isoformat() if user.created_at else None,
            'last_login': user.last_login.isoformat() if user.last_login else None,
            'is_active': user.is_active,
            'email_verified': user.email_verified,
            'exported_at': datetime.utcnow().isoformat()
        }
        
        # Log data export for audit
        current_app.logger.info(f"Data export requested for user {user_id}")
        
        return jsonify({
            'message': 'Data export successful',
            'data': user_data
        }), 200
        
    except Exception as e:
        current_app.logger.error(f"Data export error: {e}")
        return jsonify({
            'error': 'Failed to export data'
        }), 500
